OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -254,12 +254,39 @@ private static void HandleSuffixPronoun(ref StringBuilder formattedString, ReadO
             }
 
         }
+
+        private static void ToRoman(ref StringBuilder formattedString, ReadOnlySpan<DreamValue> interps, int nextInterpIndex, bool upperCase) {
+            char[] arr;
+            if(upperCase) {
+                arr = new char[] { 'M', 'D', 'C', 'L', 'X', 'V', 'I' };
+            } else {
+                arr = new char[] { 'm', 'd', 'c', 'l', 'x', 'v', 'i' };
+            }
+
+            int[] numArr = new int[] { 1000, 500, 100, 50, 10, 5, 1 };
+
+            if(!interps[nextInterpIndex].TryGetValueAsInteger(out int value)) {
+                return;
+            }
+
+            var i = 0;
+            while(value != 0) {
+                if(value >= numArr[i]) {
+                    value -= numArr[i];
+                    formattedString.Append(arr[i]);
+                } else {
+                    i++;
+                }
+            }
```

Comment:
This needs to handle negative values.

While BYOND doesn't nicely handle infinity here, I think we should.